### PR TITLE
Added TLS support for gangway

### PR DESCRIPTION
**Pull request by carldanley**
This PR adds the option to enable serving HTTPS via TLS certs.

### CODE REVIEW

**Comment by carldanley** [comment]
cc: @craigtracey

File: cmd/gangway/config.go [hunk 1]
```diff
@@ -27,26 +27,32 @@ type Config struct {
...
+	AuthorizeURL    string   `yaml:"authorizeURL" envconfig:"authorize_url"`
+	TokenURL        string   `yaml:"tokenURL" envconfig:"token_url"`
+	ClientID        string   `yaml:"clientID" envconfig:"client_id"`
+	ClientSecret    string   `yaml:"clientSecret" envconfig:"client_secret"`
+	Audience        string   `yaml:"audience"`
+	RedirectURL     string   `yaml:"redirectURL" envconfig:"redirect_url"`
+	Scopes          []string `yaml:"scopes"`
+	UsernameClaim   string   `yaml:"usernameClaim" envconfig:"username_claim"`
+	EmailClaim      string   `yaml:"emailClaim" envconfig:"email_claim"`
+	UseTLS          bool     `yaml:"useTLS" envconfig:"use_tls"`
```
Comment (craigtracey) [inline cmd/gangway/config.go]: Let's collapse this into a condition that checks for cert and key being present. `UseTLS` is not necessary in that case.

File: cmd/gangway/config.go [hunk 2]
```diff
@@ -27,26 +27,32 @@ type Config struct {
...
+	TLSCertFilePath string   `yaml:"tlsCertFilePath" envconfig:"tls_cert_file_path"`
```
Comment (craigtracey) [inline cmd/gangway/config.go]: Let's make these less verbose.  Perhaps `certFile` and `keyFile`?

**Comment by carldanley** [comment]
@craigtracey r4r again ;)

**Comment by carl-danley-heptio** [comment]
@craigtracey Made the changes we discussed in Slack (about reverting back to basics) - please let me know if anything else needs to be changed.

**Comment by carldanley** [comment]
@craigtracey Made the changes we discussed in Slack (about reverting back to basics) - please let me know if anything else needs to be changed.

**Pull request merged**